AssetRipper.Library/Exporters/Audio/AudioMixerExportCollection.cs

```diff
@@ -0,0 +1,306 @@
+using AssetRipper.Core;
+using AssetRipper.Core.Classes.Misc;
+using AssetRipper.Core.Classes.Object;
+using AssetRipper.Core.Interfaces;
+using AssetRipper.Core.Logging;
+using AssetRipper.Core.Parser.Files.SerializedFiles;
+using AssetRipper.Core.Project;
+using AssetRipper.Core.Project.Collections;
+using AssetRipper.Core.Project.Exporters;
+using AssetRipper.Core.Utils;
+using AssetRipper.SourceGenerated.Classes.ClassID_241;
+using AssetRipper.SourceGenerated.Classes.ClassID_243;
+using AssetRipper.SourceGenerated.Classes.ClassID_244;
+using AssetRipper.SourceGenerated.Classes.ClassID_245;
+using AssetRipper.SourceGenerated.Subclasses.AudioMixerConstant;
+using AssetRipper.SourceGenerated.Subclasses.AudioMixerGroupView;
+using AssetRipper.SourceGenerated.Subclasses.EffectConstant;
+using AssetRipper.SourceGenerated.Subclasses.ExposedAudioParameter;
+using AssetRipper.SourceGenerated.Subclasses.GroupConstant;
+using AssetRipper.SourceGenerated.Subclasses.GUID;
+using AssetRipper.SourceGenerated.Subclasses.Parameter;
+using AssetRipper.SourceGenerated.Subclasses.PPtr_AudioMixerEffectController_;
+using AssetRipper.SourceGenerated.Subclasses.PPtr_AudioMixerSnapshot_;
+using AssetRipper.SourceGenerated.Subclasses.SnapshotConstant;
+using AssetRipper.SourceGenerated.Subclasses.Utf8String;
+using Cpp2IL.Core;
+using System.Collections.Generic;
+using System.Linq;
+using Logger = AssetRipper.Core.Logging.Logger;
+
+namespace AssetRipper.Library.Exporters.Audio
+{
+	public class AudioMixerExportCollection : AssetsExportCollection
+	{
+		public AudioMixerExportCollection(IAssetExporter assetExporter, VirtualSerializedFile virtualFile,
+			IAudioMixerController mixer) : base(assetExporter, mixer)
+		{
+			AssetsProcessingContext context = new()
+			{
+				mixer = mixer,
+				constants = mixer.MixerConstant_C241,
+				virtualFile = virtualFile,
+				indexToGuid = new(),
+			};
+			ProcessAssets(ref context);
+		}
+
+		private struct AssetsProcessingContext
+		{
+			public IAudioMixerController mixer;
+			public IAudioMixerConstant constants;
+			public VirtualSerializedFile virtualFile;
+			public Dictionary<uint, GUID> indexToGuid;
+			public IAudioMixerGroupController[] groups;
+		}
```

Comment:
I would prefer groups be made into a List, and have this be a `private readonly record struct`.